app/service/collection.py

```diff
@@ -109,24 +107,26 @@ def validate_multiple_ids(
 
 
 @validate_call(config=ConfigDict(arbitrary_types_allowed=True))
-def validate(import_ids: set[str], db: Session = db_session.get_db()) -> None:
+def validate(import_ids: set[str], db: Optional[Session]) -> None:
```

Comment:
FYI: The function signatures for all services were changed so that `get_db` was not called when the module was loaded ... this is deferred to inside the function if `None`